shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -940,6 +950,34 @@ - (BOOL)isAlwaysUse24HourFormat {
   return [dateFormat rangeOfString:@"a"].location == NSNotFound;
 }
 
+- (NSString*)brightnessMode {
+  if (@available(iOS 13, *)) {
+    UIUserInterfaceStyle style = self.traitCollection.userInterfaceStyle;
+
+    if (style == UIUserInterfaceStyleDark) {
+      return @"dark";
+    } else {
+      return @"light";
+    }
+  } else {
+    return @"light";
+  }
+}
+
+- (NSString*)contrastMode {
+  if (@available(iOS 13, *)) {
+    UIAccessibilityContrast contrast = self.traitCollection.accessibilityContrast;
+
+    if (contrast == UIAccessibilityContrastHigh) {
```

Comment:
This constant only exists in the iOS 13 SDK, unlike UIUserInterfaceStyleDark which is in 12. This isn't going to compile on the bots.

If you want to add this we need to go back to the early discussion of declaring the constant yourself, conditionally, behind an ifdef based on the compiling SDK, so that you can use the constant regardless of which SDK you're compiling against. Same with adding a declaration of accessibilityContrast, which is also 13+

Reply:
Do you have an example of defining such a constant that I can reference and replicate?

Reply:
I took a stab at only including the accessibility logic when the max API is 13 or above. Please let me know if that's sufficient.

Reply:
As previously discussed, putting the actual implementation behind ifdefs doesn't do what you want, because that would mean it would only work if someone compiles their own engine locally with the Xcode beta; none of this code would actually be shipped in our library.

If this code is important to have now, then per the previous discussion what you need are declarations of the constant and the method that are behind reversed SDK ifdefs (i.e., the local declarations are used when compiling with a stable SDK, and not there to cause duplicate definition errors when using the beta SDK), allowing all of the logic here to compile regardless of SDK.

I don't have an example offhand, and would have to dig one up tomorrow. But the declarations that you need behind the ifdefs are exactly the ones that are in the beta SDK: the same name and value for the constant, and the identical signature for the method (declared via a class extension).

Reply:
An example to replicate would be great. I wasn't clear on the distinction last time and I haven't been able to locate an example of it in practice.

Reply:
Here are some examples; they are old and on macOS, but the principle is exactly the same.
method: https://hg.mozilla.org/camino/file/d974e1a3e6f55d09feb5afdf5ee50aa97f846518/src/browser/BrowserWindowController.mm#l108
constant: https://hg.mozilla.org/camino/rev/3249ec97333c#l3.12